Auth/Auth.Platform.Android/Credentials/OIDCCredentialManager.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using AeroGear.Mobile.Auth.Credentials;
+using Android.App;
+using Android.Content;
+using Java.Lang;
+
+namespace AeroGear.Mobile.Auth.Credentials
+{
+    /// <summary>
+    /// Class for peristing, retrieving, updating and removing OpenID Connect
+    /// credentials on a device.
+    /// </summary>
+    public class OIDCCredentialManager : ICredentialManager
+    {
+        private static readonly string StoreName = "AeroGear.Mobile.Auth.Credentials";
+
+        private ISharedPreferences SharedPreferences;
+
+        private static OIDCCredentialManager _Instance;
+
+        /// <summary>
+        /// Instance of the credential manager using the application context.
+        /// </summary>
+        /// <value>Instance of <see cref="OIDCCredentialManager"/></value>
+        public static OIDCCredentialManager Instance
+        {
+            get
+            {
+                if (_Instance == null)
+                {
+                    _Instance = new OIDCCredentialManager(Application.Context.ApplicationContext);
+                }
+                return _Instance;
+            }
+        }
+
+        private OIDCCredentialManager(Context context)
+        {
+            SharedPreferences = context.GetSharedPreferences(StoreName, FileCreationMode.Private);
```

Comment:
@aidenkeating The current implementation doesn't seem like we need to use stuff like `NSCoable`. But if it turns out we need, we can always add a platform-specific implementation. We will try our best to use the same code across platform, but it may not be always the case.

Reply:
@wei-lee Ok cool, it makes sense now. Thanks